echopop/computation/biology.py

```diff
@@ -104,160 +128,751 @@ def sum_strata_weight( haul_data: pd.DataFrame ,
+    )
+    
+    # ---- Calculate aggregate sums of fish weights
+    specimen_binned_weight = (
+        specimen_data_filtered
+        # ---- Group weight summations across stratum/species/sex/length/age
+        .groupby( [ 'stratum_num' , 'species_id' , 'sex' , 'length_bin' , 'age_bin' ] )
+        # ---- Sum the weights 
+        .agg( weight_all =( 'weight' , 'sum' )  , weight_adult =( 'weight_adult' , 'sum' ) ) 
+        # ---- Fill empty/non-existent values with 0's
+        .fillna( 0 )
         .reset_index( )
-        .replace( np.nan , 0 )
-        .assign( total_weight_sex_all = lambda df: df.groupby( [ 'stratum_num' , 'sex' ] )[ 'summed_weight_all' ].transform( sum ) ,
-                 total_weight_sex_adult = lambda df: df.groupby( [ 'stratum_num' , 'sex' ] )[ 'summed_weight_adult' ].transform( sum ) ,
-                 proportion_weight_all = lambda df: df.summed_weight_all / df.total_weight_sex_all ,
-                 proportion_weight_adult = lambda df: df.summed_weight_adult / df.total_weight_sex_adult )            
+    )
+    
+    ### Calculate the relative weight proportions of each length-age bin for each sex within each stratum
+    proportions_weight_length_age_sex = (
+        specimen_binned_weight
+        # ---- Calculate total sex-specific weights for each stratum
+        .assign( total_weight_sex_all = lambda df: df.groupby( [ 'stratum_num' , 'species_id' , 'sex'  ] )[ 'weight_all' ].transform( sum ) ,
+                 total_weight_sex_adult = lambda df: df.groupby( [ 'stratum_num' , 'species_id' , 'sex' ] )[ 'weight_adult' ].transform( sum ) )
+        # ---- Calculate the weight proportions within each sex: from Matlab --> Len_age_key_wgt_*n
+        .assign( proportion_weight_sex_all = lambda df: df.weight_all / df.total_weight_sex_all ,
+                 proportion_weight_sex_adult = lambda df: df.weight_adult / df.total_weight_sex_adult )
     )
 
-    ### Carriage return
-    return weight_strata , weight_strata_station , weight_length_age_sex_stratum
+    ### Fill empty/non-existent values with 0's
+    proportions_weight_length_age_sex[ 'proportion_weight_sex_all' ] = (
+        proportions_weight_length_age_sex[ 'proportion_weight_sex_all' ].fillna( 0 )
+    )
+
+    proportions_weight_length_age_sex[ 'proportion_weight_sex_adult' ] = (
+        proportions_weight_length_age_sex[ 'proportion_weight_sex_adult' ].fillna( 0 )
+    )
 
-def normalize_length_age_sex_weight_proportions( weight_length_age_sex_stratum: pd.DataFrame ,
-                                                 weight_strata: pd.DataFrame ):
+    ### Return output
+    return proportions_weight_length_age_sex
+
+def compute_aged_sex_weight_proportions( proportions_weight_length_age_sex: pd.DataFrame ,
+                                         aged_proportions: pd.DataFrame ):
+    """
+    Compute the aged proportions across all fish and specific sexes for each
+    length and age bin  
```

Comment:
This function doesn't actually get into the length and age bin, that is in the previous function I think.
```suggestion
def compute_aged_sex_weight_proportions( proportions_weight_length_age_sex: pd.DataFrame ,
                                         aged_proportions: pd.DataFrame ):
    """
    Compute the weight proportion for aged/unaged fish of either sex (2x2 choice) across all fish.
    Also compute the weight proportion of only adult aged/unaged fish of either sex across all fish.
```